### PR TITLE
Centralize cache service instances creation in tests

### PR DESCRIPTION
**Pull request by tlrx**
This pull request centralizes the creation of `CacheService` instances in Searchable Snapshot tests. It cleans up a bit the tests and it will make the tests more easily adaptable when the creation of `CacheService` will be more complex due to the future persistent cache.

### CODE REVIEW

File: x-pack/plugin/searchable-snapshots/src/test/java/org/elasticsearch/xpack/searchablesnapshots/AbstractSearchableSnapshotsTestCase.java [hunk 1]
```diff
@@ -0,0 +1,138 @@
...
+        recoveryState.setStage(RecoveryState.Stage.FINALIZE).setStage(RecoveryState.Stage.DONE);
+
+        return recoveryState;
+    }
+
+    /**
+     * Wait for all operations on the threadpool to complete
+     */
+    protected static void assertThreadPoolNotBusy(ThreadPool threadPool) throws Exception {
+        assertBusy(() -> {
```
Comment (original-brownbear) [inline x-pack/plugin/searchable-snapshots/src/test/java/org/elasticsearch/xpack/searchablesnapshots/AbstractSearchableSnapshotsTestCase.java:131]: The method this replaces in some spots used a 30s timeout. When I created it I just copied those 30s from previous code so I don't know if we actually need the long of a timeout here, but maybe we do and should put I back here?

**Review by original-brownbear — APPROVED** [review]
LGTM :)

File: x-pack/plugin/searchable-snapshots/src/test/java/org/elasticsearch/xpack/searchablesnapshots/AbstractSearchableSnapshotsTestCase.java (hunk 1 above)
Comment (tlrx) [reply x-pack/plugin/searchable-snapshots/src/test/java/org/elasticsearch/xpack/searchablesnapshots/AbstractSearchableSnapshotsTestCase.java:131]: Thanks for catching this. I looked a bit at the history and I think it's preferable to use a long timeout here (mostly for testRandomReads())

**Comment by tlrx** [comment]
Thanks Armin!

**Pull request merged**